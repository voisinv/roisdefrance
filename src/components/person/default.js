//@flow

import type { Person, PersonNode, PersonLink } from '../../types/person';

const PersonDefault = (): Person => {
  return {
    value: '',
    children: []
  };
};

const PersonNodeDefault = (): PersonNode => {
  return {
    data: PersonDefault(),
    x: 0,
    y: 0,
    children: [],
    depth: 0
  };
};

const PersonLinkDefault = ():PersonLink => {
  return {
    source: PersonNodeDefault(),
    target: PersonNodeDefault()
  }
};

export {
  PersonDefault,
  PersonNodeDefault,
  PersonLinkDefault
};