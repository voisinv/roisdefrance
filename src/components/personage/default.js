//@flow

import type { Personage, PersonageNode, PersonageLink } from '../../types/personage';

const PersonageDefault = (): Personage => {
  return {
    value: '',
    children: []
  };
};

const PersonageNodeDefault = (): PersonageNode => {
  return {
    data: PersonageDefault(),
    x: 0,
    y: 0,
    children: [],
    depth: 0
  };
};

const PersonageLinkDefault = (): PersonageLink => {
  return {
    source: PersonageNodeDefault(),
    target: PersonageNodeDefault()
  }
};

export {
  PersonageDefault,
  PersonageNodeDefault,
  PersonageLinkDefault
};