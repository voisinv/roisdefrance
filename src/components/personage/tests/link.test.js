import React from 'react';
import renderer from 'react-test-renderer';

import LinkContainer from '../link';
import {Link} from '../Link';
import { PersonageLinkDefault, PersonageNodeDefault } from '../default';
describe('Link', () => {
  it('renders correctly', () => {
    const props = {
      link: PersonageLinkDefault(),
      configuration: {
        colors: '#567899'
      }
    };
    const element = renderer.create(<Link {...props}></Link>);
    expect(element).toMatchSnapshot();
  });
});
describe('LinkContainer', () => {
  describe('', () => {
    let source, target, link, actions, element;
    beforeEach(() => {
      source = Object.assign({}, PersonageNodeDefault(), { x: 140, y: 150 });
      target = Object.assign({}, PersonageNodeDefault(), { x: 160, y: 170 });
      link = Object.assign({}, { source, target });
      actions = {colors: () => ''};
      element = renderer.create(<LinkContainer link={link} actions={actions}/>).toJSON();
    });

    it('renders correctly', () => {
      expect(element).toMatchSnapshot();
    });
    it('match properties', () => {
      expect(element).toBeTruthy();
      expect(element.type).toEqual('line');

      expect(element.props.x1).toBeDefined();
      expect(element.props.x1).toEqual(140);

      expect(element.props.y1).toBeDefined();
      expect(element.props.y1).toEqual(150);

      expect(element.props.x2).toBeDefined();
      expect(element.props.x2).toEqual(160);

      expect(element.props.y2).toBeDefined();
      expect(element.props.y2).toEqual(170);;
    });
  });

  it('renders a default link', () => {
    const actions = {colors: () => ''};
    const line = renderer.create(<LinkContainer actions={actions}/>).toJSON();

    expect(line).toBeTruthy();
    expect(line.type).toEqual('line');

    expect(line.props.x1).toBeDefined();
    expect(line.props.x1).toEqual(0);

    expect(line.props.y1).toBeDefined();
    expect(line.props.y1).toEqual(0);

    expect(line.props.x2).toBeDefined();
    expect(line.props.x2).toEqual(0);

    expect(line.props.y2).toBeDefined();
    expect(line.props.y2).toEqual(0);
  });
});
