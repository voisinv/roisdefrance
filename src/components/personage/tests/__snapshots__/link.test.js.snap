// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<line
  style={
    Object {
      "opacity": 0.7,
      "stroke": undefined,
      "strokeDasharray": 10,
      "strokeWidth": 2,
    }
  }
  x1={0}
  x2={0}
  y1={0}
  y2={0}
/>
`;

exports[`LinkContainer  renders correctly 1`] = `
<line
  style={
    Object {
      "opacity": 0.7,
      "stroke": undefined,
      "strokeDasharray": 10,
      "strokeWidth": 2,
    }
  }
  x1={140}
  x2={160}
  y1={150}
  y2={170}
/>
`;
